src/noui.cpp

```diff
@@ -53,7 +56,39 @@ void noui_InitMessage(const std::string& message)
 
 void noui_connect()
 {
-    uiInterface.ThreadSafeMessageBox_connect(noui_ThreadSafeMessageBox);
-    uiInterface.ThreadSafeQuestion_connect(noui_ThreadSafeQuestion);
-    uiInterface.InitMessage_connect(noui_InitMessage);
+    noui_ThreadSafeMessageBoxConn = uiInterface.ThreadSafeMessageBox_connect(noui_ThreadSafeMessageBox);
+    noui_ThreadSafeQuestionConn = uiInterface.ThreadSafeQuestion_connect(noui_ThreadSafeQuestion);
+    noui_InitMessageConn = uiInterface.InitMessage_connect(noui_InitMessage);
+}
+
+bool noui_ThreadSafeMessageBoxSuppressed(const std::string& message, const std::string& caption, unsigned int style)
+{
+    return false;
+}
+
+bool noui_ThreadSafeQuestionSuppressed(const std::string& /* ignored interactive message */, const std::string& message, const std::string& caption, unsigned int style)
+{
+    return false;
```

Comment:
Instead of ignoring the message in the test, shouldn't the test check for the correct messag?

Reply:
While that's a nice feature to have, this isn't being done currently as well. It would require a much bigger change to achieve that, giving the opportunity to the caller of noui_supress to supply an expected error, or noui_suppress to retain whatever messages are being thrown to it, and adding a method to retrieve them in the test and do assertions on it. I would however prefer to do this in an additional PR on top of this one.

Reply:
Right now at least the test runner can check for the correct reason by reading the stdour (or err). You take that option without providing an alternative.

Reply:
Is that feature used somewhere?

If not I suggest that feature can be added when needed.

Would be nice to get this merged.

Reply:
I don't think i'm taking any option away: The suppressing only happens when you explicitly call `noui_suppress()`. So the default is to still print it to stdout, which could be useful if a test needs it. The places that are producing the unwanted output were not using stdout and i'm providing a way for those tests to selectively suppress it.

Reply:
I don't think there is a use case in suppressing the output with no option to read it

Reply:
The output is not being checked as it stands. If not checking the output is a problem in itself, then that deserves a fix _too_, but is not the scope of what I was aiming for with this PR.